utils/swift_build_support/swift_build_support/products/cmake_product.py

```diff
@@ -132,3 +134,645 @@ def install_with_cmake(self, install_targets, install_destdir):
+                                       self.args.darwin_deployment_version_watchos)
+            swift_cmake_options.define('CMAKE_OSX_SYSROOT:PATH', cmake_os_sysroot)
+            swift_cmake_options.define('CMAKE_OSX_ARCHITECTURES', arch)
+
+            # lldb_cmake_options+=(
+            #     -DCMAKE_OSX_DEPLOYMENT_TARGET:STRING="${cmake_osx_deployment_target}"
+            #     -DCMAKE_OSX_SYSROOT:PATH="${cmake_os_sysroot}"
+            #     -DCMAKE_OSX_ARCHITECTURES="${architecture}"
+            # )
+            # llbuild_cmake_options+=(
+            #     -DCMAKE_OSX_ARCHITECTURES="${architecture}"
+            # )
+            # ;;
+
+        # # We don't currently support building compiler-rt for cross-compile targets.
+        # # It's not clear that's useful anyway.
+        # if [[ $(is_cross_tools_host "${host}") ]] ; then
+        #   llvm_cmake_options+=(
+        #       -DLLVM_TOOL_COMPILER_RT_BUILD:BOOL=FALSE
+        #       -DLLVM_BUILD_EXTERNAL_COMPILER_RT:BOOL=FALSE
+        #   )
+        # else
+        #   llvm_cmake_options+=(
+        #       -DLLVM_TOOL_COMPILER_RT_BUILD:BOOL="$(
+        #                         false_true ${SKIP_BUILD_COMPILER_RT})"
+        #       -DLLVM_BUILD_EXTERNAL_COMPILER_RT:BOOL="$(
+        #                         false_true ${SKIP_BUILD_COMPILER_RT})"
+        #   )
+        # fi
+
+        if self.is_cross_compile_target(host_target):
+            llvm_cmake_options.define('LLVM_TOOL_COMPILER_RT_BUILD:BOOL', 'FALSE')
+            llvm_cmake_options.define('LLVM_BUILD_EXTERNAL_COMPILER_RT:BOOL', 'FALSE')
+        else:
+            # TODO: is build-script-impl wrong here and these should be negated?
+            llvm_cmake_options.define('LLVM_TOOL_COMPILER_RT_BUILD:BOOL',
+                                      cmake.CMakeOptions.true_false(
+                                          self.args.build_compiler_rt))
+            llvm_cmake_options.define('LLVM_BUILD_EXTERNAL_COMPILER_RT:BOOL',
+                                      cmake.CMakeOptions.true_false(
+                                          self.args.build_compiler_rt))
+
+        # # If we are asked to not generate test targets for LLVM and or Swift,
+        # # disable as many LLVM tools as we can. This improves compile time when
+        # # compiling with LTO.
+        # #
+        # # *NOTE* Currently we do not support testing LLVM via build-script. But in a
+        # # future commit we will.
+        # #for arg in "$(compute_cmake_llvm_tool_disable_flags)"; do
+        # #    llvm_cmake_options+=( ${arg} )
+        # #done
+
+        # if [[ "${llvm_target_arch}" ]] ; then
+        #     llvm_cmake_options+=(
+        #         -DLLVM_TARGET_ARCH="${llvm_target_arch}"
+        #     )
+        # fi
+
+        if llvm_target_arch:
+            llvm_cmake_options.define('LLVM_TARGET_ARCH', llvm_target_arch)
+            llvm_cmake_options.define('LLVM_DEFAULT_TARGET_TRIPLE', swift_host_triple)
```

Comment:
This is not in `build-script-impl` and is redundant, because [`LLVM_HOST_TRIPLE` below is used to initialize `LLVM_DEFAULT_TARGET_TRIPLE` anyway](https://github.com/apple/llvm-project/blob/7b73ca3043fecfc5fa6bbf5b28edfee61a83ff9f/llvm/CMakeLists.txt#L687).